apps/ledger-live-desktop/src/newArch/features/ModularDrawer/Web3AppWebview/__tests__/openAssetAndAccountDrawer.test.ts

```diff
@@ -0,0 +1,92 @@
+import { setDrawer } from "~/renderer/drawers/Provider";
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import {
+  openAssetAndAccountDrawer,
+  openAssetAndAccountDrawerPromise,
+} from "../AssetAndAccountDrawer";
+
+jest.mock("~/renderer/drawers/Provider", () => ({
+  setDrawer: jest.fn(),
+}));
+
+const mockAccount = { id: "account1" } as AccountLike;
+const mockParentAccount = { id: "parent1" } as Account;
+
+describe("openAssetAndAccountDrawer", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Mock setDrawer to simulate the behavior of onAccountSelected
+    (setDrawer as jest.Mock).mockImplementation((_Component, props, options) => {
+      if (props?.onAccountSelected) {
+        props.onAccountSelected(mockAccount, mockParentAccount);
+      }
+      if (options?.onRequestClose) {
+        options.onRequestClose();
+      }
+    });
+  });
+
+  it("should handle callback mode success", () => {
+    const onSuccess = jest.fn();
+    openAssetAndAccountDrawer({
+      onSuccess,
+    });
+
+    const call = (setDrawer as jest.Mock).mock.calls[0];
+    const onAccountSelected = call[1].onAccountSelected;
+
+    onAccountSelected(mockAccount, mockParentAccount);
+
+    expect(onSuccess).toHaveBeenCalledWith(mockAccount, mockParentAccount);
+  });
+
+  it("should handle callback mode cancel", () => {
+    const onCancel = jest.fn();
+
+    openAssetAndAccountDrawer({
+      onCancel,
+    });
+
+    const call = (setDrawer as jest.Mock).mock.calls[0];
+    const onRequestClose = call[2].onRequestClose;
+
+    onRequestClose();
+
+    expect(onCancel).toHaveBeenCalled();
+  });
+});
+
+describe("openAssetAndAccountDrawerPromise", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Mock setDrawer to simulate the behavior of onAccountSelected and onRequestClose
+    (setDrawer as jest.Mock).mockImplementation((_Component, props, options) => {
+      if (props?.onAccountSelected) {
+        props.onAccountSelected(mockAccount, mockParentAccount);
+      }
+      if (options?.onRequestClose) {
+        options.onRequestClose();
+      }
```

Comment:
The first test (openAssetAndAccountDrawer) fails in the CI. It's probably an issue with the test. I think you could simplify the debugging by changing the `beforeEach` of both `openAssetAndAccountDrawer` and `openAssetAndAccountDrawerPromise` similarly to what you wrote here but something like this:
```suggestion
      selectAccount.mockImplementation(props?.onAccountSelected);
      cancel.mockImplementation(options?.onRequestClose);
```

With `selectAccount` and `cancel` defined in the root scope:
```ts
const selectAccount = jest.fn();
const cancel = jest.fn();
```
Then in the 2 success tests you can call `selectAccount(mockAccount, mockParentAccount)` and the 2 cancell tests `cancel()`